falcon/media/json.py

```diff
@@ -11,21 +12,70 @@ class JSONHandler(BaseHandler):
 
     This handler uses Python's :py:mod:`json` by default, but will
     use :py:mod:`ujson` if available.
+
+    Keyword Arguments:
+        dumps (func): the function to use when serializing JSON responses.
+        loads (func): the function to use when deserializing JSON requests.
+
+    You can override the JSON library used by changing the ``dumps`` and
+    ``loads`` functions. Some good options are orjson, python-rapidjson,
+    and mujson.::
+
+        import falcon
+        from falcon import media
+
+        import rapidjson
+
+        json_handler = media.JSONHandler(
+            dumps=rapidjson.dumps,
+            loads=rapidjson.loads,
+        )
+        extra_handlers = {
+            'application/json': json_handler,
+        }
+
+        api = falcon.API()
+        api.req_options.media_handlers.update(extra_handlers)
+        api.resp_options.media_handlers.update(extra_handlers)
+
+    By default, ``ensure_ascii`` is passed to the ``json.dumps`` function.
+    If you override the ``dumps`` function, you will need to explicitly set
+    ``ensure_ascii`` to ``False`` in order to enable the serialization of
+    Unicode characters to UTF-8. This is easily done by using
+    ``functools.partial`` to curry the keyword arguments. This gives you
```

Comment:
As said in a comment before, I misled you with this *curry*. Apparently, it has a slightly different meaning than partial application.